Tests/MessagingInApp/MessagingInAppImplementationTest.swift

```diff
@@ -237,3 +250,18 @@ class MessagingInAppImplementationTest: UnitTest {
         XCTAssertEqual(inAppProviderMock.dismissMessageCallsCount, 1)
     }
 }
+
+extension MessagingInAppImplementationTest {
+    func postEventAndWait<E: EventRepresentable>(event: E, timeoutInSeconds: Double = 1.0) {
+        eventBusHandler.postEvent(event)
+        waitForEventBus(timeoutInSeconds: timeoutInSeconds)
+    }
+
+    func waitForEventBus(timeoutInSeconds: Double = 1.0) {
+        let expectation = XCTestExpectation(description: "wait for \(timeoutInSeconds) seconds")
+        DispatchQueue.main.asyncAfter(deadline: .now() + timeoutInSeconds) {
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: timeoutInSeconds)
+    }
```

Comment:
This eventbus wait function looks like a sleep function (wait X amount of time, hope that the operation completes in that time). 

I strongly want to avoid introducing a sleep pattern in our test suite to avoid our test suite running longer then necessary to run. 

Is there a way that we can have the `XCTestExpectation` only wait for the eventbus to finish? I am assuming I would see code like this: 

```swift
    func postEventAndWait<E: EventRepresentable>(event: E, timeoutInSeconds: Double = 1.0) {
        eventBusHandler.postEvent(event)
        let expectEventBusToFinishPosting = XCTestExpectation
        eventBusHandler.postAllEvents(onComplete: {
          expectEventBusToFinishPosting.fulfill()
        })

        waitfor(expectEventBusToFinishPosting)
    }
```

Is this possible? If so, is it within the scope of this PR or do we need a new ticket for it?

Reply:
Unfortunately, no. [`EventBus`](https://github.com/customerio/customerio-ios/blob/main-replica-for-cdp/Sources/Common/Communication/EventBusHandler.swift#L100-L110) does not have anything today that we can utilize to confirm the event emission. However, if we want to update it, we can create separate ticket for it and update other EventBus tests in it too.

Reply:
I had a feeling that we may need to add this functionality to the EventBus. 

Thanks for pointing out that the EventBus already has some other tests today that would also need changed. 

Because of this scope, I agree, creating a separate ticket would be a good move. [I made one here](https://linear.app/customerio/issue/MBL-95/reduce-time-it-takes-for-test-suite-to-execute-by-replacing-sleep).

Reply:
Thanks for creating the ticket